django/views/generic/detail.py

```diff
@@ -159,12 +168,14 @@ def get_template_names(self):
                     object_meta.model_name,
                     self.template_name_suffix
                 ))
-            elif hasattr(self, 'model') and self.model is not None and issubclass(self.model, models.Model):
-                names.append("%s/%s%s.html" % (
-                    self.model._meta.app_label,
-                    self.model._meta.model_name,
-                    self.template_name_suffix
-                ))
+            elif hasattr(self, 'get_model'):
```

Comment:
I'm not sure how much of an issue it is but this will be backward incompatible with `SingleObjectTemplateResponseMixin` subclasses that define a `model` attribute but don't have a `get_model()` method. There is no such class in Django's provided CBVs.